### PR TITLE
feat(snap): add initial snap packaging

### PR DESCRIPTION
**Pull request by MonicaisHer**
fix: #6 
Signed-off-by: Mengyi <mengyi.wang@canonical.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-snmp-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) no hardware available
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) this will be done after the snap been published
  <link to docs PR>

## Test guideline:
```
git clone https://github.com/MonicaisHer/device-snmp-go.git
cd device-snmp-go
git checkout add-snap-v2
snapcraft
snap install edgexfoundry --channel=2.0
snap install edgex-device-snmp_2.0.1-dev.4_amd64.snap --dangerous
sudo snap connect edgexfoundry:edgex-secretstore-token edgex-device-snmp:edgex-secretstore-token
sudo nano /var/snap/edgex-device-snmp/current/config/device-snmp/res/profiles/device.snmp.trendnet.TPE082WS.yaml
```
please change the deviceResources's RebootCurrentState's properties readWrite: "R" to "RW", to let device profile align the device resource. (issue: #126 )
#### check if the snap is running:
```
snap start --enable edgex-device-snmp
snap logs edgex-device-snmp
```
expect:
```
level=INFO ts=2021-10-28T11:38:56.615380966Z app=device-snmp source=message.go:55 msg="device snmp started"
```
#### ping edgex-device-snmp's port:
```
curl http://localhost:59993/api/v2/ping
```
expect:{"apiVersion":"v2","timestamp":"Wed Oct 27 14:11:36 CEST 2021"}

### CODE REVIEW

**Comment by keanjapesan** [comment]
recheck

File: snap/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+The `device-config` content interface allows another snap to seed this device snap with both a configuration file and one or more device profiles.
+
+To use, create a new snap with a directory containing the configuration and device profile files. Your snapcraft.yaml file then needs to define a slot with read access to the directory you are sharing.
+
+```
+slots:
+  device-config:
+    interface: content  
+    content: device-config
+    write:
```
Comment (farshidtz) [inline snap/README.md:82]: The key should be `read`, not `write` as it is exposing the read-only `$SNAP/config` directory.

File: snap/README.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+**Note** - at this time changes to configuration values in the [Writable] section are not supported. For details on the mapping of configuration options to Config options, please refer to "Service Environment Configuration Overrides".
+
+## Service Environment Configuration Overrides
+
+**Note** - all of the configuration options below must be specified with the prefix: `env.`
+
+```
+[Service]
+service.boot-timeout            // Service.BootTimeout
```
Comment (farshidtz) [inline snap/README.md:142]: Probably a deprecated config argument. Not supported by the snap.

File: snap/README.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+service.health-check-interval   // Service.HealthCheckInterval
+service.host                    // Service.Host
+service.server-bind-addr        // Service.ServerBindAddr
+service.port                    // Service.Port
+service.protocol                // Service.Protocol
```
Comment (farshidtz) [inline snap/README.md:147]: `Service.Protocol` is probably a deprecated config argument. Not supported by the snap.

File: snap/README.md (hunk 1 above)
Comment (MonicaisHer) [reply snap/README.md]: Thank you for your re-review, I have updated this part. I also updated this part's word based on your another [PR](https://github.com/edgexfoundry/device-mqtt-go/pull/311/files).

File: snap/README.md (hunk 2 above)
Comment (MonicaisHer) [reply snap/README.md]: Removed.

File: snap/README.md (hunk 3 above)
Comment (MonicaisHer) [reply snap/README.md]: Removed.

**Review by farshidtz — APPROVED** [review]
Thanks. Looks good.
Please squash and merge which should also fix the semantic commit issues.

**Comment by MonicaisHer** [comment]
@siggiskulason  Could you please help me squash and merge this PR? Thank you!

**Review by siggiskulason — APPROVED** [review]
LGTM

**Pull request merged**